src/ocean_emulators/models/modules/blocks.py

```diff
@@ -121,7 +123,10 @@ def forward(self, fts: torch.Tensor) -> torch.Tensor:
                 fts = torch.nn.functional.pad(
                     fts, (0, 0, self.N_pad, self.N_pad), mode="constant"
                 )
-            fts = layer(fts)
+            if self.checkpoint_simple and not isinstance(layer, nn.Conv2d):
+                fts = torch.utils.checkpoint.checkpoint(layer, fts, use_reentrant=False)
+            else:
+                fts = layer(fts)
```

Comment:
This arrangement of conditionals makes the most sense to me -- what do you think?

```suggestion
            elif self.checkpoint_simple:
                fts = torch.utils.checkpoint.checkpoint(layer, fts, use_reentrant=False)
            else:
                fts = layer(fts)
```

Reply:
This is not equivalent -- in your updated code the layer is never invoked for Conv2D. I have a slight preference for keeping in the way it is over duplicating the line `fst = layer(fts)` in the first if block. WDYT?

Reply:
I'm a little confused -- in the original code, isn't it also never invoked for Conv2D? isn't that what the `not isinstance(layer, nn.Conv2d)` does in the `if` statement?

Reply:
oh! In the else block!

Reply:
Ok, I follow and agree the original way is better.